cpp/arcticdb/async/async_store.hpp

```diff
@@ -22,227 +22,263 @@ namespace arcticdb::async {
 template<class ClockType = util::SysClock>
 class AsyncStore : public Store {
 public:
-    AsyncStore(
-        std::shared_ptr<storage::Library> library,
-        const arcticdb::proto::encoding::VariantCodec &codec) :
-        library_(std::move(library)),
-        codec_(new arcticdb::proto::encoding::VariantCodec{codec}) {
+    AsyncStore(std::shared_ptr<storage::Library> library, const arcticdb::proto::encoding::VariantCodec& codec)
+        : library_(std::move(library)),
+          codec_(new arcticdb::proto::encoding::VariantCodec{codec})
+    {
     }
 
-    folly::Future<entity::VariantKey> write(
-        stream::KeyType key_type,
+    folly::Future<entity::VariantKey> write(stream::KeyType key_type,
         VersionId version_id,
         const StreamId& stream_id,
         IndexValue start_index,
         IndexValue end_index,
-        SegmentInMemory &&segment) override {
+        SegmentInMemory&& segment) override
+    {
 
         util::check(segment.descriptor().id() == stream_id,
-                    "Descriptor id mismatch in atom key {} != {}",
-                    stream_id,
-                    segment.descriptor().id());
-
-        return async::submit_cpu_task(EncodeAtomTask{
-            key_type, version_id, stream_id, start_index, end_index, current_timestamp(),
-            std::move(segment), library_, codec_, size_t(0)
-        })
+            "Descriptor id mismatch in atom key {} != {}",
+            stream_id,
+            segment.descriptor().id());
+
+        return async::submit_cpu_task(EncodeAtomTask{key_type,
+                                          version_id,
+                                          stream_id,
+                                          start_index,
+                                          end_index,
+                                          current_timestamp(),
+                                          std::move(segment),
+                                          library_,
+                                          codec_,
+                                          size_t(0)})
```

Comment:
I really like the original two-line wrapping for this call. When I read that, my brain is able to parse the structure quickly and I can then focus on the important bits.

The long wrapping, on the other hand, robs the focus and I have to triple read to make the connection between `submit_cpu_task` and `via` below.

Reply:
There are rare occasions where we don't want to enforce wrapping. If we don't enforce it it will dissolve into anarchy. On the other hand I feel like having multiple things on the same line is confusing and easy to misread.